### PR TITLE
[mlir][tensor] Fix tensor::PackOp fold() handling of padding value

### PR DESCRIPTION
**Pull request by hanhanW**
We can't just check if it is a splat constant or not. We should also check if the value match.

### CODE REVIEW

File: mlir/lib/Dialect/Tensor/IR/TensorOps.cpp [hunk 1]
```diff
@@ -4143,9 +4145,12 @@ bool PackOp::isLikePad() {
 }
 
 OpFoldResult PackOp::fold(FoldAdaptor adaptor) {
+  std::optional<Attribute> paddingValue;
+  if (adaptor.getPaddingValue())
+    paddingValue = adaptor.getPaddingValue();
```
Comment (joker-eph) [inline mlir/lib/Dialect/Tensor/IR/TensorOps.cpp:4150]: Nit: we should avoid the double call to `getPaddingValue()` here: this accessor is far from free (there is a loop hidden behind it)

File: mlir/lib/Dialect/Tensor/IR/TensorOps.cpp [hunk 2]
```diff
@@ -1069,9 +1069,11 @@ void EmptyOp::getCanonicalizationPatterns(RewritePatternSet &results,
 /// Try to remove a tensor operation if it would only reshape a constant.
 /// Removes the op and replaces the constant with a new constant of the result
 /// shape.
-static OpFoldResult reshapeConstantSource(DenseElementsAttr source,
-                                          TensorType result) {
-  if (source && source.isSplat() && result.hasStaticShape())
+static OpFoldResult
+reshapeConstantSource(DenseElementsAttr source, TensorType result,
+                      std::optional<Attribute> cst = std::nullopt) {
```
Comment (joker-eph) [inline mlir/lib/Dialect/Tensor/IR/TensorOps.cpp:1074]: Please update the documentation for this.

**Review by chelini — APPROVED** [review]
It looks good to me after addressing Mehdi's comments.

**Comment by chelini** [comment]
FYI: @rengolin

**Review by joker-eph — APPROVED** [review]

**Pull request merged**